api/checks/update.go

```diff
@@ -115,30 +100,42 @@ func updateCheckHandler(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
-func getDiffSummary(ctx context.Context, before, after map[string][]int, checkState summaries.CheckState) summaries.Summary {
+func getDiffSummary(aeAPI shared.AppEngineAPI, diffAPI shared.DiffAPI, before, after shared.TestRun) (summaries.Summary, error) {
 	diffFilter := shared.DiffFilterParam{Added: true, Changed: true, Unchanged: true}
-	diff := shared.GetResultsDiff(before, after, diffFilter, nil, nil)
+	diff, err := diffAPI.GetRunsDiff(before, after, diffFilter, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	product, _ := shared.ParseProductSpec(before.Product.String())
+	checkState := summaries.CheckState{
+		Product:    product,
+		HeadSHA:    before.FullRevisionHash,
+		Title:      getCheckTitle(product),
+		DetailsURL: diffAPI.GetMasterDiffURL(before.FullRevisionHash, product),
+		Status:     "in_progress",
```

Comment:
IIUC, this is the final step of this custom check, where we surface regression to GitHub, so should this be "completed" instead of "in_progress"?

Reply:
Ha, good catch ( must have been like this already ). 
```suggestion
		Status:     "completed",
```